docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,409 @@
+# Semantic Conventions for Messaging Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Producer metrics](#producer-metrics)
+  * [Metric: `messaging.publish.duration`](#metric-messagingpublishduration)
+  * [Metric: `messaging.publish.messages`](#metric-messagingpublishmessages)
+- [Consumer metrics](#consumer-metrics)
```

Comment:
I think it would be helpful to explain how the time from a message being fetched from the broker to being settled on the broker is divided up. For example, if the client fetches a bunch of messages they might be processed progressively, some of them being temporarily enqueued in the client until they're dispatched for processing. Where does this time waiting for dispatch get accounted for, if at all?

Reply:
absolutely, created https://github.com/open-telemetry/semantic-conventions/issues/218 to design observability story for it.

Reply:
This is an important observability tasks, however, I'd strongly suggest to leave this out of the scope of v1.0 of messaging semantics. As with traces, I'd be in favor of focusing on operations embedded in the user application code (deliver callbacks, receive operations), and in a next step (post v1) go deeper and observe what the client SDKs do under the hood (pre-fetching before deliver calls, or during receive calls).